sdk/python/kfp/dsl/component_spec_test.py

```diff
@@ -105,14 +105,10 @@ def test_build_component_spec_from_structure(self):
           'is_root_component': True,
           'expected_result': {
               'inputDefinitions': {
-                  'artifacts': {
-                      'input1': {
-                          'artifactType': {
-                              'schemaTitle': 'system.Dataset'
```

Comment:
This is an example related to my other comment. If user explicit type annotated an input as 'Dataset', but provide a string value. We should probably better throw an error ask them to correct the typing instead of changing this for them.

Reply:
TBH, This example was pretty strange and it wasn't runnable in the IR implementations.

This example change is caused more by the removal of automatic Importer insertion.
I'll change the types accordingly, so that this does not look weird.

Reply:
Right, this won't work right now. It was possible partially because of the auto-injected importer.

Reply:
AFAIK, the IR does not support artifact inputs in the root components. The ComponentSpec in the previous version of the test was invalid. This change at least makes the spec valid.

Perhaps we should change this test.

Reply:
The Dataset input with constant argument has been removed from the test case.